atlp-rwanda/e-comm-team-techtitans-bn
# #184752256 feature(view-a-specific-product): enable a user to view a specific product

# What is this PR for?
This PR brings up the feature that enables a user to display a specific product. This happens when he/she clicks on one item in a catalog or another list of products, then it is displayed alone with more details.
It also includes all undone tests from previous tasks (signup & updating product availability status).

# How to locally test if it works

- Open your vs-code, switch to this branch manually (not via the terminal) and look for `origin/ft-view-a-specific-product-#184752256`. 
- After making sure you are in that specific branch, pull the latest changes. 
- Run the script `npm run dev`
- Now open Postman and: signup, login (as a seller), verifyOtp, create a product category, then create a Product.
-  Using a `GET` request, use this URL [http://localhost:3000/product/:productId](http://localhost:3000/product/:productId)
- `:productId` at the end of the product-endpoint indicates the product's id of the product you specifically want to view. Copy and paste it, replacing `:productId` at the end of the endpoint.
- It should now return the product successfully.

# Screenshots 
## Postman, Swagger Documentation, and Tests
<img width="628" alt="Screen Shot 2023-04-18 at 19 36 19" src="https://user-images.githubusercontent.com/108181471/232858712-7b24f4e5-3f75-49fe-ae4e-ffd3d56714c4.png">
<img width="1411" alt="Screen Shot 2023-04-18 at 19 40 27" src="https://user-images.githubusercontent.com/108181471/232859620-bfaddf1f-03aa-4907-983d-dc0d62b3b2e3.png">
<img width="728" alt="Screen Shot 2023-04-18 at 19 44 06" src="https://user-images.githubusercontent.com/108181471/232860364-14016f7a-5242-4b1c-ba3c-65c47d8fa7ef.png">


## NB:
If the above steps didn't work, feel free to leave a comment of what is not working or a suggestion of what needs to be changed.
For more info, please take a look at the [Pivotal tracker story](https://www.pivotaltracker.com/story/show/184752256) details.

`Thanks`.
